docs/source/quickstart.rst

```diff
@@ -89,18 +89,18 @@ And update functions of the trainer and evaluator are simply:
     def _update(engine, batch):
         model.train()
         optimizer.zero_grad()
-        x, y = _prepare_batch(batch, cuda)
+        x, y = _prepare_batch(batch, requires_grad=False, device=device)
```

Comment:
@sean-adler here should not be `requires_grad=True` ?

Reply:
I think the batch shouldn't require gradient tracking - if it did then `optimizer.step()` would update `x` and `y`, and if I make the following change:

```diff
    def _update(engine, batch):
        model.train()
        optimizer.zero_grad()
-       x, y = _prepare_batch(batch, requires_grad=False, device=device)
+       x, y = _prepare_batch(batch, requires_grad=True, device=device)
        loss = loss_fn(y_pred, y)
        loss.backward()
        optimizer.step()
        return loss.item()
```

Running `python examples/mnist.py --epochs=1` in my local environment raises this error:

```
Traceback (most recent call last):
  File "examples/mnist.py", line 98, in <module>
    run(args.batch_size, args.val_batch_size, args.epochs, args.lr, args.momentum, args.log_interval)
  File "examples/mnist.py", line 78, in run
    trainer.run(train_loader, max_epochs=epochs)
  File "/Users/sean/anaconda3/envs/test-environment/lib/python3.5/site-packages/ignite-0.1.0a1-py3.5.egg/ignite/engines/engine.py", line 151, in run
  File "/Users/sean/anaconda3/envs/test-environment/lib/python3.5/site-packages/ignite-0.1.0a1-py3.5.egg/ignite/engines/engine.py", line 117, in _handle_exception
  File "/Users/sean/anaconda3/envs/test-environment/lib/python3.5/site-packages/ignite-0.1.0a1-py3.5.egg/ignite/engines/engine.py", line 138, in run
  File "/Users/sean/anaconda3/envs/test-environment/lib/python3.5/site-packages/ignite-0.1.0a1-py3.5.egg/ignite/engines/engine.py", line 111, in _run_once_on_dataset
  File "/Users/sean/anaconda3/envs/test-environment/lib/python3.5/site-packages/ignite-0.1.0a1-py3.5.egg/ignite/engines/engine.py", line 117, in _handle_exception
  File "/Users/sean/anaconda3/envs/test-environment/lib/python3.5/site-packages/ignite-0.1.0a1-py3.5.egg/ignite/engines/engine.py", line 101, in _run_once_on_dataset
  File "/Users/sean/anaconda3/envs/test-environment/lib/python3.5/site-packages/ignite-0.1.0a1-py3.5.egg/ignite/engines/__init__.py", line 34, in _update
  File "/Users/sean/anaconda3/envs/test-environment/lib/python3.5/site-packages/torch/tensor.py", line 93, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
  File "/Users/sean/anaconda3/envs/test-environment/lib/python3.5/site-packages/torch/autograd/__init__.py", line 89, in backward
    allow_unreachable=True)  # allow_unreachable flag
RuntimeError: No grad accumulator for a saved leaf!
```

I'm pretty sure the issue is that `y` is a `torch.LongTensor`, and you can't backprop to that data type in 0.4.0. Seems like it's the same issue as https://github.com/pytorch/pytorch/issues/7021 (which is not a bug).

Reply:
@sean-adler I agree, so should not we just change `_prepare_batch` as :
```
def _prepare_batch(batch, requires_grad=False, device=None):
    x, y = batch
    return convert_tensor(x, requires_grad=requires_grad, device=device), 
               convert_tensor(y, requires_grad=False, device=device)
```
what do you think ?

Reply:
I don't see why `x` would need to be updated either, but I could definitely be missing something - I'm happy to make the change if you think it's worth it.

Reply:
I checked an example  in 0.3.1 with `Variable` etc and it seems, you're right, we do not need `requires_grad` to be true for `x` neither. So we can simply remove it from the signature of 
`def _prepare_batch(batch, requires_grad=False, device=None)`...